http.go

```diff
@@ -1,17 +1,142 @@
 package veneur
 
 import (
+	"compress/zlib"
+	"encoding/json"
+	"hash/fnv"
+	"io"
 	"net/http"
+	"sort"
+	"time"
 
+	"github.com/Sirupsen/logrus"
 	"goji.io"
 	"goji.io/pat"
 	"golang.org/x/net/context"
 )
 
 func (s *Server) Handler() http.Handler {
 	mux := goji.NewMux()
+
 	mux.HandleFuncC(pat.Get("/healthcheck"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte("ok\n"))
 	})
+
+	mux.HandleFuncC(pat.Post("/import"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		var (
+			jsonMetrics []JSONMetric
+			body        io.ReadCloser
+			err         error
+		)
+		switch enc := r.Header.Get("Content-Encoding"); enc {
+		case "":
+			body = r.Body
+		case "deflate":
```

Comment:
Should we add a tag for `response_duration_ns` to note compressed versus uncompressed? Might not be very useful for us, but might be for others.